### ISSUE TITLE
Issue on Linear Motion File in Sloshing tank example

### ISSUE THREAD

**Issue by VictorV666** [DualSPHysics/DesignSPHysics#98]
Hello, 

I would like to give a heads-up of a potential bug when I am running a 2D case of a tank with water inside (sloshing case), under a linear motion file - horizontal translation (a text file which has 2 columns). When I define the X-column as 1, and time as 0, it runs, but when I vizualize actually it translatez on Z-direction, and the particles are out of the boundary. I tried to reverse them, and use Z as my X, but it just translates endlessly, without actually following the time steps created on the text file that I input as linear motion file (I attached the external file). If other files are required, please let me know. I would be interested if this is a general problem, or it is just by me, since I don't get errors on that.

The version used for FreeCad is 0.19 with DualSPHysics package: 0.6.1.2102-24-04.
 
[motion_a5mm_f01-08Hz.txt](https://github.com/DualSPHysics/DesignSPHysics/files/6609655/motion_a5mm_f01-08Hz.txt)

 Thank you!

**Comment by zlotny** [comment]
Could you upload the case xml file? I wonder if this problem is related with DualSPHysics. Maybe DualSPHysics can read your txt file correctly. I'll try to test this to see where the problem is.

Thanks for noticing it!

**Comment by VictorV666** [comment]
Sure, please find it attached!
[slosh_Def.zip](https://github.com/DualSPHysics/DesignSPHysics/files/7945632/slosh_Def.zip)

**Comment by zlotny** [comment]
Hi there, I've been trying different things with your file and there's something weird with it. Then I tested with a newly made file so I can give you a working recommendation. These are the problems I see with it:
1. Don't use tabs. Theres multiple tabs at the end of each line that can confuse DualSPHyiscs. I would process that file so it only separates values by 1 space each (like: "0 0.2" and so on).
2. Create another column with 0 values. This way you can set the coordinates you don't want to move to that column. That was one of the main issues. For example if you want the object only to move on the X axis you could have a file in which each line was something like "1 0.2 0", being 1 second of time, an space, 0.2 is the value for an axis, and 0 is just a null value. Then in DesignSPHysics you could use it like this:

![image](https://user-images.githubusercontent.com/6155245/151790404-25be7f56-71d1-404d-99bb-fae2753c2bf4.png)

Notice how Y and Z point to the last column (2, beginning in column 0) so they won't move, but X is pointing to the middle column (1, beginning in column 0) which has the actual values I want

3. The movement in the file is reeeeeally tiny. Take in mind the values are measured in seconds and in meters. It may lead to confusion that the "standard" unit in FreeCAD by default is mm. For I can see you're moving your half a meter long tank by 0.004 meters (or 4 mm) when the time hits 3 second. Are you sure this is correct?
 
4. Try to avoid exponentials. This is not specifically bad but sometimes different softwares use different notations for exponentials in text files, and I'm not really sure which format is the preferred one for DualSPHysics. Sure, the file will be bigger but in a text file nowadays... it's not a big deal at least you plan on moving it via diskette :)


I hope this helps, I'll keep the issue open for now, but the main problem seems solvable with a few lines of python or matlab. 

Have a nice day!

**Comment by VictorV666** [comment]
Hi, thank you for taking the time to answer! I am still in doubt as I checked a case, with defining the motion (ex: sinusoidal rectilinear motion) - I will snap the images to explain - basically I want to give a sinusoidal harmonic of 0.5Hz and 5cm amplitude on X-dir, but by looking at the paraview - the particles are just dispersed on Z dir. This did not happen in the previous, older DSPH versions. I attach also the input files - i used GUI for that. 
![motion_sloshing](https://user-images.githubusercontent.com/80780895/152856794-d95eb540-a21b-4d97-993f-1f1983537caa.PNG)
![visual](https://user-images.githubusercontent.com/8078089
[DSPH_Case.zip](https://github.com/DualSPHysics/DesignSPHysics/files/8018065/DSPH_Case.zip)
5/152856799-316175ab-d5af-4a50-acec-d583386c0787.PNG)

Thanks and best regards,
Victor

**Comment by VictorV666** [comment]
Dear Andres,

I figured out why the above was happening - namely, the decimal symbol on my windows was set to comma (","), not point (".") , the same for digit decimal (it should be a comma). After I fixed that, it works. A small detail, but it took some time to figure it out, and I decided to post it here for the users who might experience this issue in the future. With that, I would like to close the issue, as solved. Thank you!

Best regards,
Victor
